### PR TITLE
allow increasing total promised in a single lock

### PR DESCRIPTION
**Pull request by Guillembonet**
We got the value, added, and then saved it. This could potentially lead to a broken state (though it's rare). Now we will allow increasing the value in a single lock to avoid this.

Signed-off-by: Guillem Bonet <guillem@mysterium.network>

### CODE REVIEW

File: session/pingpong/consumer_totals_storage.go [hunk 1]
```diff
@@ -85,6 +87,47 @@ func (cts *ConsumerTotalsStorage) Get(chainID int64, id identity.Identity, herme
...
+	if err != nil {
+		// wrap the error to an error we can check for
+		if errors.Is(storm.ErrNotFound, err) {
+			log.Debug().Msg("No previous invoice grand total, assuming zero")
+			value = big.NewInt(0)
+		} else {
+			return fmt.Errorf("could not get total promised to add : %w", err)
+		}
+	}
+	if value == nil || value.Cmp(big.NewInt(0)) == -1 {
```
Comment (tomasmik) [inline session/pingpong/consumer_totals_storage.go:107]: Can it even be `nil`? https://go.dev/play/p/I5VFOAy5_kD

Comment (Guillembonet) [reply session/pingpong/consumer_totals_storage.go:107]: fun fact: [yes](https://go.dev/play/p/_PH7DtQ2jed)

**Comment by Snawoot** [comment]
@n10ty take a look as well, please

**Review by Snawoot — APPROVED** [review]

**Review by n10ty — CHANGES_REQUESTED** [review]
This changes is useless until someone using same ID for the same node instance, which we are not supporting now.

**Pull request closed without merging**